easybuild/tools/modules.py

```diff
@@ -1104,17 +1104,22 @@ def load(self, modules, mod_paths=None, purge=False, init_env=None, allow_reload
             # restore initial environment if provided
             if init_env is None:
                 raise EasyBuildError("Initial environment required when purging before loading, but not available")
-            else:
-                restore_env(init_env)
 
-            # make sure $MODULEPATH is set correctly after purging
-            self.check_module_path()
+            restore_env(init_env)
+            # sync mod_paths member with MODULEPATH environment variable
+            self.set_mod_paths()
```

Comment:
`check_module_path` does more than `set_mod_paths` does, so I think we need to be careful that we're not introducing a regression here...

This mainly matters when the path in which EasyBuild is creating modules is not in the original `$MODULEPATH`.
That's fine, because EasyBuild will add this location to `$MODULEPATH` via the logic in `check_module_path` when `self.mod_paths` is set to `None`, after calling `set_mod_paths`.

So, a more correct fix would be to set `self.mod_paths` to `None`, and then call `self.check_module_paths`.

```suggestion
            self.mod_paths = None
            self.check_module_path()
```

Reply:
fixed in https://github.com/xdelaruelle/easybuild-framework/pull/2

I'm confident that this is the right way to fix this bug, also verified that `test_load_module` which was enhanced to capture this bug still passes when using `EnvironmentModules` as modules tool (which should be confirmed by CI)